static/app/components/externalIssues/ticketRuleModal.tsx

```diff
@@ -79,6 +79,9 @@ export default function TicketRuleModal({
   const queryClient = useQueryClient();
   const api = useApi({persistInFlight: true});
   const organization = useOrganization();
+  // The instance are values from the saved rule. Once a user modifies the form, we don't want to
+  // override any inputs with these instance values.
+  const [showInstanceValues, setShowInstanceValues] = useState(true);
```

Comment:
Model is really complicated, but essentially persisting the values could lead to us not saving them when the new inputs rerender. Yes it means that writing text, and changing a dynamic field will cause them to disappear -- but it's better than thinking they will save because we render it and then it not actually saving because it wasn't persisted in state.

It also prevents a situation where the value becomes stale because the async option isn't applicable under the new form (e.g. assigned to X, in project X, but when switched to project Y, X isn't a member but we persist the assignee)